Description: Run math_test on x87
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: https://github.com/google/highway/issues/1708
Last-Update: 2023-09-13

Index: highway/hwy/contrib/math/math_test.cc
===================================================================
--- highway.orig/hwy/contrib/math/math_test.cc
+++ highway/hwy/contrib/math/math_test.cc
@@ -66,6 +66,10 @@ namespace HWY_NAMESPACE {
 #define HWY_MATH_TEST_EXCESS_PRECISION 0
 #endif  // HWY_ARCH_X86_32 etc
 
+#if HWY_MATH_TEST_EXCESS_PRECISION == 1
+#error invalid setup
+#endif
+
 template <class T, class D>
 HWY_NOINLINE void TestMath(const char* name, T (*fx1)(T),
                            Vec<D> (*fxN)(D, VecArg<Vec<D>>), D d, T min, T max,
